### ISSUE TITLE
Add support for building DATABASE_URL from parts in Docker again

### ISSUE THREAD

**Issue by kodebach** [kimai/kimai#5218]
### Describe the problem

With the old tobybatch/kimai2 Docker image you could use `DB_USER`, `DB_PASS`, etc. instead of the single `DATABASE_URL` environment variable. That was really handy because it meant you could separate the password from the rest of the parameters.

For example, you could read `DB_PASS` from the `.env` file in Compose and add `.env` to `.gitignore` to avoid leaking the password via git. It also allowed using Docker/Podman secrets for the password.

The parsing of `DATABASE_URL` in `entrypoint.sh` also breaks for some IPv6 addresses in the host part, because those contain `:`. 

### Describe the solution you'd like

Restore the logic from here https://github.com/tobybatch/kimai2/blob/85319d49ea1bb1b02e554c7764e422320facfeb7/assets/startup.sh#L7-L43 to replace https://github.com/kimai/kimai/blob/17a815e5a9550b301a073caf49882a4436a1c647/.docker/entrypoint.sh#L6-L25

Or at least, add an option that allows setting the password via a separate variable.

### Describe alternatives you've considered

My current workaround is to use

```
/bin/env "DATABASE_URL=mysql://kimai:${DB_PASS}@mysql/kimai?charset=utf8mb4&serverVersion=8.3.0" /entrypoint.sh
```

 as the command/entrypoint for the docker container. 

In case anybody stumbles upon this, to make this work in Quadlet you actually need to specify it like this:

```
[Container]
Image=docker.io/kimai/kimai2:apache
Exec=/bin/bash -c '/bin/env "DATABASE_URL=mysql://kimai:$${DB_PASS}@host.containers.internal/kimai?charset=utf8mb4&serverVersion=8.3.0" /entrypoint.sh'
Secret=kimai-db-password,type=env,target=DB_PASS
```

**Comment by kevinpapst** [comment]
Why?
You can use a .env file or whatever the equivalet might be in that Podman
```
KIMAI_USER=kimai@example.com
KIMAI_PASSWORD=password

MYSQL_DATABASE=kimai
MYSQL_USER=kimai
MYSQL_PASSWORD=kimai
MYSQL_ROOT_PASSWORD=kimai
```
and in docker-compose.yaml
```
      environment:
          - ADMINMAIL=${KIMAI_USER}
          - ADMINPASS=${KIMAI_PASSWORD}
          - "DATABASE_URL=mysql://${MYSQL_USER}:${MYSQL_PASSWORD}@sqldb/${MYSQL_DATABASE}?charset=utf8mb4&serverVersion=8.3.0"
          - TRUSTED_HOSTS=localhost,127.0.0.1,${SUBDOMAIN}.${DOMAIN_NAME}
          - TZ=${TIMEZONE}
      restart: unless-stopped
```
For the full example see https://github.com/kimai/kimai-docker-caddy/blob/main/.env

I never used IPv6 adresses for internal traffic. Can you send a PR to fix the parsing?

**Comment by kodebach** [comment]
> You can use a .env file or whatever the equivalet might be in that Podman [...]

You're right, that would work for Compose, if you keep the password in an `.env` file. But it doesn't address the use case of Docker/Podman secrets, because those are directly injected into the container (as files or environment variables), you can't use them to build environment variables in Compose.

Also, I'm actually using Quadlet not Compose, so I need to specify the `DATABASE_URL` as an `Environment=` key in a systemd file and there I can't use a separate `.env` file and AFAIK you can't build env variables from other env variables anyway.

> I never used IPv6 adresses for internal traffic. Can you send a PR to fix the parsing?

I'm not using them either. I just noticed that the script splits on `:` and that would break IPv6. I'm also not actually sure how to fix the parsing while keeping it as simple as it is. AFAIK `:` is allowed in URIs only in the host part, if that starts with `[` and ends with `]`, i.e. a IPv6 address as host. Properly parsing that in a shell script seems rather annoying. 

However, it's probably not a big issue anyway, since you can just use `--add-host`/`extra_hosts` to add a hostname for the IPv6 address. Might be easier to just document that IPv6 is not supported directly and you need to define a hostname if you need IPv6 support.
